qiita_pet/static/js/networkVue.js

```diff
@@ -742,79 +705,75 @@ Vue.component('processing-graph', {
-          randomSeed: 1,
-          improvedLayout: true,
-          hierarchical: {
-            direction: "LR",
-            sortMethod : 'directed',
-            levelSeparation: 190,
-            shakeTowards: 'roots'
-          }
-        },
-        interaction: {
-          dragNodes: false,
-          dragView: true,
-          zoomView: true,
-          selectConnectedEdges: true,
-          navigationButtons: true,
-          keyboard: false
-        },
-        groups: {
-          jobs: {
-            color: '#FF9152'
-          },
-          artifact: {
-            color: '#FFFFFF'
-          },
-          type: {
-            color: '#BBBBBB'
-          }
-        }
+      var style = [{
+        selector: 'node',
+        style: {
+          'content': 'data(label)',
+          'background-color': 'data(color)',
+          'shape': 'data(shape)',
+          'text-opacity': 0.7,
+          'text-wrap': "wrap",
+          'border-color': '#BBBBBB',
+          'border-width': '1px'
+        }}, {
+        selector: 'edge',
+        style: {
+          'curve-style': 'bezier',
+          'target-arrow-shape': 'triangle'
+        }},
+      ];
+      var panzoom_options =	{
+        zoomOnly: true,
+        sliderHandleIcon: 'fa fa-minus',
+        zoomInIcon: 'fa fa-plus',
+        zoomOutIcon: 'fa fa-minus',
+        resetIcon: 'fa fa-expand'
       };
 
-      vm.network = new vis.Network(container, data, options);
-      vm.network.on("stabilized", function (params) {
+      // Note: we only need to sort the edges to keep the same structure of the
+      //       graph; in other words, nodes order is not important
+      vm.edges = vm.edges.sort((a, b) => (a.data.source > b.data.source) ? 1 : (a.data.source === b.data.source) ? ((a.data.target > b.data.target) ? 1 : -1) : -1 );
+
+      vm.network = cytoscape({
+          container: container,
+          minZoom: 1e-50,
+          maxZoom: 2,
+          layout: layout, style: style,
+          elements: {
+            nodes: vm.nodes,
+            edges: vm.edges,
+          }
+        });
+      vm.network.panzoom(panzoom_options);
+      vm.network.nodes().lock();
+
+      vm.network.ready(function() {
         if (vm.new_job_info !== null){
-          vm.network.focus(vm.new_job_info["job_id"], {
-            scale: vm.new_job_info["scale"]
-          });
+          vm.network.viewport(vm.new_job_info['viewport']);
           vm.new_job_info = null;
         }
       });
 
-      vm.network.on("click", function (properties) {
-        var ids = properties.nodes;
-        if (ids.length == 0) {
-          return
-        }
-        // [0] cause users can only select 1 node
-        var clickedNode = vm.nodes_ds.get(ids)[0];
-        var element_id = ids[0];
-        if (clickedNode.group == 'artifact') {
+      vm.network.on('tap', 'node', function (evt) {
+        var data = evt.target.data();
+        var element_id = data.id;
+
+        if (data.group == 'artifact') {
```

Comment:
minor i think here, but possible to use `===` to be consistent with the other equalities?